src/main/java/edu/harvard/iq/dataverse/dataaccess/DataAccess.java

```diff
@@ -35,29 +33,33 @@ public DataAccess() {
     }
 
     // set by the user in glassfish-setup.sh if DEFFAULT_STORAGE_DRIVER_IDENTIFIER = swift
+    // or DEFFAULT_STORAGE_DRIVER_IDENTIFIER = s3
     public static final String DEFAULT_STORAGE_DRIVER_IDENTIFIER = System.getProperty("dataverse.files.storage-driver-id");
     
-    // The getDataFileIO() methods initialize DataFileIO objects for
+    // The getStorageIO() methods initialize StorageIO objects for
     // datafiles that are already saved using one of the supported Dataverse
     // DataAccess IO drivers.
-    public static DataFileIO getDataFileIO(DataFile df) throws IOException {
-        return getDataFileIO(df, null);
+    public static <T extends DvObject> StorageIO<T> getStorageIO(T dvObject) throws IOException {
+        return getStorageIO(dvObject, null);
     }
 
-    public static DataFileIO getDataFileIO(DataFile df, DataAccessRequest req) throws IOException {
-
-        if (df == null
-                || df.getStorageIdentifier() == null
-                || df.getStorageIdentifier().equals("")) {
+    //passing DVObject instead of a datafile to accomodate for use of datafiles as well as datasets
+    public static <T extends DvObject> StorageIO<T> getStorageIO(T dvObject, DataAccessRequest req) throws IOException {
+        
+        if (dvObject == null
+                || dvObject.getStorageIdentifier() == null
+            || dvObject.getStorageIdentifier().isEmpty()) {
             throw new IOException("getDataAccessObject: null or invalid datafile.");
         }
 
-        if (df.getStorageIdentifier().startsWith("file://")
-                || (!df.getStorageIdentifier().matches("^[a-z][a-z]*://.*"))) {
-            return new FileAccessIO (df, req);
-        } else if (df.getStorageIdentifier().startsWith("swift://")){
-            return new SwiftAccessIO(df, req);
-        } else if (df.getStorageIdentifier().startsWith("tmp://")) {
+        if (dvObject.getStorageIdentifier().startsWith("file://")
+                || (!dvObject.getStorageIdentifier().matches("^[a-z][a-z0-9]*://.*"))) {
+            return new FileAccessIO<>(dvObject, req);
+        } else if (dvObject.getStorageIdentifier().startsWith("swift://")){
+            return new SwiftAccessIO<>(dvObject, req);
+        } else if (dvObject.getStorageIdentifier().startsWith("s3://")){ //fixme: change name?
```

Comment:
Should this fixme be removed? Did we change the name from `s3://` to something else?